### ISSUE TITLE
Disallowed tags inside of List of Figures in UA-1 export mode, according to veraPDF

### ISSUE THREAD

**Issue by SeSodesa** [typst/typst#7054]
### Description

My thesis template ([in-progress GitLab merge request](https://gitlab.com/tuni-official/thesis-templates/tau-typst-thesis-template/-/merge_requests/83)) currently compiles with the version of Typst on `main`, but when running it through veraPDF I get the following XML report:
```xml
<?xml version="1.0" encoding="utf-8"?>
<report>
  <buildInformation>
    <releaseDetails id="core" version="1.28.2" buildDate="2025-07-15T16:07:00+03:00"></releaseDetails>
    <releaseDetails id="validation-model" version="1.28.2" buildDate="2025-07-15T16:11:00+03:00"></releaseDetails>
    <releaseDetails id="gui" version="1.28.0" buildDate="2025-07-15T17:44:00+03:00"></releaseDetails>
  </buildInformation>
  <jobs>
    <job>
      <item size="555221">
        <name>/Users/soderhos/tau-typst-thesis-template/template/main.pdf</name>
      </item>
      <validationReport jobEndStatus="normal" profileName="PDF/UA-1 validation profile" statement="PDF file is not compliant with Validation Profile requirements." isCompliant="false">
        <details passedRules="104" failedRules="2" passedChecks="127630" failedChecks="3">
          <rule specification="ISO 14289-1:2014" clause="7.2" testNumber="19" status="failed" failedChecks="1" tags="structure,list">
            <description>L element may contain only L, LI and Caption elements</description>
            <object>SEL</object>
            <test>kidsStandardTypes.split('&amp;').filter(elem =&gt; elem != 'L' &amp;&amp; elem != 'LI' &amp;&amp; elem != 'Caption').length == 0 || kidsStandardTypes == ''</test>
            <check status="failed">
              <context>root/document[0]/StructTreeRoot[0](39 0 obj PDStructTreeRoot)/K[0](85 0 obj SEDocument Document)/K[29](1116 0 obj SEL L)</context>
              <errorMessage>L element contains Span,Span,Span,Formula,Span,Span,Formula,Span
,Span,Span,Span,Span,Span,Span,Span,Formula,Span,Span element(s) instead of L, LI or Caption</errorMessage>
            </check>
          </rule>
          <rule specification="ISO 14289-1:2014" clause="7.7" testNumber="1" status="failed" failedChecks="2" tags="structure,alt-text">
            <description>All mathematical expressions shall be enclosed within a Formula tag as detailed in ISO 32000-1:2008, 14.8.4.5 and shall have Alt or ActualText attributes</description>
            <object>SEFormula</object>
            <test>(Alt != null &amp;&amp; Alt != '') || ActualText != null</test>
            <check status="failed">
              <context>root/document[0]/StructTreeRoot[0](39 0 obj PDStructTreeRoot)/K[0](85 0 obj SEDocument Document)/K[67](846 0 obj SEP P)/K[0](849 0 obj SEFormula Formula)</context>
              <errorMessage>Formula structure element neither has an alternate description nor a replacement text</errorMessage>
            </check>
            <check status="failed">
              <context>root/document[0]/StructTreeRoot[0](39 0 obj PDStructTreeRoot)/K[0](85 0 obj SEDocument Document)/K[78](734 0 obj SEP P)/K[1](735 0 obj SEFormula Formula)</context>
              <errorMessage>Formula structure element neither has an alternate description nor a replacement text</errorMessage>
            </check>
          </rule>
        </details>
      </validationReport>
      <duration start="1759908843005" finish="1759908843646">00:00:00.641</duration>
    </job>
  </jobs>
  <batchSummary totalJobs="1" failedToParse="0" encrypted="0" outOfMemory="0" veraExceptions="0">
    <validationReports compliant="0" nonCompliant="1" failedJobs="0">1</validationReports>
    <featureReports failedJobs="0">0</featureReports>
    <repairReports failedJobs="0">0</repairReports>
    <duration start="1759908842969" finish="1759908843655">00:00:00.686</duration>
  </batchSummary>
</report>
```
Related to the invalid list (L) element structure with spans and such inside of it, when looking at the output PDF in a plain text editor, the offending element seems to be the list of figures created [here](https://gitlab.com/tuni-official/thesis-templates/tau-typst-thesis-template/-/blob/b0.17.0/tauthesis.typ#L1577) or list of listings created [here](https://gitlab.com/tuni-official/thesis-templates/tau-typst-thesis-template/-/blob/b0.17.0/tauthesis.typ#L1611). I also have a show rule related to displaying outline elements [here](https://gitlab.com/tuni-official/thesis-templates/tau-typst-thesis-template/-/blob/b0.17.0/tauthesis.typ#L693), which might be contributing to the problem.

Not sure what to do in userland to prevent this from happening. Figure captions might contain inline equations, so this is not something that can be easily avoided. Any tips would be appreciated.

### Reproduction URL

https://gitlab.com/tuni-official/thesis-templates/tau-typst-thesis-template/-/merge_requests/83

### Operating system

macOS

### Typst version

- [x] I am using the latest version of Typst

**Comment by SeSodesa** [comment]
The PDF file is here: [main.pdf](https://github.com/user-attachments/files/22761747/main.pdf).

**Comment by saecki** [comment]
Thanks for the extensive report! Could you double check whether the linked PR fixes your issues?

**Comment by SeSodesa** [comment]
I'll look into it later today. Away from computer currently.

**Comment by SeSodesa** [comment]
Ok, so now I actually get a more descriptive error message:
```sh
error: PDF/UA-1 error: invalid list (L) structure
     ┌─ @preview/scholarly-tauthesis:0.17.0/tauthesis.typ:1158:1
     │
1158 │ ╭   grid(
1159 │ │     stroke: none,
1160 │ │     columns: (10%, auto, 85%),
1161 │ │     align: (top+left, top + center, top+left),
     · │
1170 │ │     }
1171 │ │   )
     │ ╰───^
     │
     = hint: list (L) may not contain division (Div)
     = hint: this is probably caused by a show rule
```
This is with the current `HEAD` of the branch `b0.17.0` of the thesis template. It just does not like the grid?

**Comment by SeSodesa** [comment]
Ah, and now it is actually in a term list and not table of contents.

**Comment by saecki** [comment]
Yes, the `grid` generates the `Div` elements, which aren't allowed l, because lists (`L`) must be strongly structured.

**Comment by SeSodesa** [comment]
Ok, I need to come up with a different means of making the columns of each entry in the term list of equal size (width).

**Comment by laurmaedje** [comment]
This would be another use case for a replacing selector that fully removes an element (in this case `terms`) from the semantic structure, thereby lifting these requirements. This has come up before in https://github.com/typst/typst/issues/6548#issuecomment-3266662892. I'll open an issue for that sometime.
